lib/browser/mixins/browser_wait.ts

```diff
@@ -26,13 +27,15 @@ export default abstract class BrowserWait extends BrowserNavigation {
                 visible: true
             }, ...args);
         } catch (err) {
-            if (err instanceof Error && err.message.match(/DOMException\:/)) { // TODO: move to a helper/wrapper
-                throw new QueryError("waitFor", `Invalid selector "${selector}".`);
-            } else {
+            if (err instanceof puppeteerErrors.TimeoutError) {
                 let errMsg;
                 if (typeof selector === 'function') errMsg = `Waiting for function to return true`;
                 else errMsg = `Waiting for element "${selector}"`;
                 throw new TimeoutError("waitFor", errMsg, timeout);
+            } else if (err instanceof Error && err.message.match(/DOMException\:/)) { // TODO: move to a helper/wrapper
+                throw new QueryError("waitFor", `Invalid selector "${selector}".`);
+            } else {
+                throw new QueryError("waitFor", "An unknown error occurred: " + err.message)
```

Comment:
Maybe this shouldn't be a QueryError (WendigoError or FatalError)

Also, maybe don't add the message `An unknown error occurred: `, it doesn't really add any info